crates/doc/src/inference.rs

```diff
@@ -1484,6 +1486,201 @@ impl Shape {
     }
 }
 
+const MAX_ROOT_FIELDS: usize = 750;
+const MAX_NESTED_FIELDS: usize = 200;
+
+impl Shape {
+    // Widen a Shape to make the provided AsNode value fit.
+    // Returns a hint if some locations might exceed their maximum allowable size.
+    // NOTE: If a particular location defines `additionalProperties` as a subschema, don't
+    // add the field from `AsNode`, instead jump straight to squashing the the field into `additionalProperties`
+    fn widen<'n, N>(&mut self, node: &'n N) -> bool
+    where
+        N: AsNode,
+    {
+        self.widen_inner(node, Location::Root)
+    }
+
+    fn widen_inner<'n, N>(&mut self, node: &'n N, loc: Location) -> bool
```

Comment:
I'm thinking that this should be `widen`, and the current `widen` removed? Passing `Location::Root` as a caller is easy enough

Reply:
Yep, good point. I had originally had more setup stuff in `widen()`. Removed it